projects/igniteui-angular/src/lib/grids/selection/selection.service.ts

```diff
@@ -420,57 +432,100 @@ export class IgxGridSelectionService {
             return;
         }
         clearPrevSelection = !this.grid.isMultiRowSelectionEnabled || clearPrevSelection;
+        if (this.grid.isPivot) {
+            this.selectPivotRowById(rowID, clearPrevSelection, event);
+            return;
+        }
+        const selectedRows = this.getSelectedRowsData();
+        const newSelection = clearPrevSelection ? [this.getRowDataById(rowID)] : this.rowSelection.has(rowID) ?
+            selectedRows : [...selectedRows, this.getRowDataById(rowID)];
+        const removed = clearPrevSelection ? selectedRows : [];
+        this.emitRowSelectionEvent(newSelection, [this.getRowDataById(rowID)], removed, event, selectedRows);
+    }
 
+    public selectPivotRowById(rowID, clearPrevSelection: boolean, event?): void {
         const selectedRows = this.getSelectedRows();
-        const newSelection = clearPrevSelection ? [rowID] : this.rowSelection.has(rowID) ?
-            selectedRows : [...selectedRows, rowID];
-        const removed = clearPrevSelection ? selectedRows : [];
-        this.emitRowSelectionEvent(newSelection, [rowID], removed, event);
+        const newSelection = clearPrevSelection ? [rowID] : this.rowSelection.has(rowID) ? selectedRows : [...selectedRows, rowID];
```

Comment:
PivotGrid supports only single selection, so maybe this line can be simpler. I could not think of a scenario where newSelection would contain more than one row.

Anyway, for Pivot I was not able to see any selection in the event arguments:

![image](https://user-images.githubusercontent.com/14248932/216931052-27c04c7e-e221-4401-903d-9da637acbf4e.png)


I used the Pivot Grid sample.

Reply:
![PivotGridSelection](https://user-images.githubusercontent.com/34240583/217231759-40cca138-58b4-4cc5-9749-74b89877b5ee.gif)